File: 09Parsing_html/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int extractTags(char string[]) {
	char ch;
	char stringAux[256] = "";

	int j = 0;
	for (int i = 0; i < strlen(string); i++) {
		if (string[i] == '\n')
			continue;
		if (string[i] == '<') {
			while(string[i] != '>')
				i++;
		} else
			stringAux[j++] = string[i];
	}
	strcpy(string, stringAux);
	return j; // the length of our new string
}

int main() {
	char inputFileName[] = ".\\pag1.html";
	char outputFileName[] = ".\\arquivoDeSaida.html";

	FILE *f1 = fopen(inputFileName, "r");
	FILE *f2 = fopen(outputFileName, "w");

	char string[256];

	while(fgets(string, 256, f1)) {
		if (extractTags(string)) {
			fprintf(f2, "%s\n", string);
		}
	}

	puts("Pronto, agora e so conferir o arquivo de saida");

	fclose(f1);
	fclose(f2);
	return 0;
}

File: 09Parsing_html/main.py
def extractTags(line: str):
	stringAux: str = ''

	isAngleOpen = False
	for ch in line:
		if ch == '\n':
			continue
		if ch == '<':
			isAngleOpen = True
			continue
		if ch == '>':
			isAngleOpen = False
			continue
		if isAngleOpen:
			continue
		else:
			stringAux += ch

	return stringAux


inputFileName = '.\\09Parsing_html\\pag1.html'
outputFileName = '.\\09Parsing_html\\arquivoDeSaida.html'

f1 = open(inputFileName, encoding='utf8')
f2 = open(outputFileName, 'w')

result: str = ''
for line in f1:
	result = extractTags(line)

	if result:
		f2.write(f'{result}\n')


f1.close()
f2.close()
